plugins/out_s3/s3.c

```diff
@@ -1243,8 +1303,27 @@ static int put_all_chunks(struct flb_s3 *ctx)
                                 chunk->create_time, buffer, buffer_size);
             flb_free(buffer);
             if (ret < 0) {
-                s3_store_file_unlock(chunk);
                 chunk->failures += 1;
+                if (ctx->has_old_buffers == FLB_FALSE) {
```

Comment:
So this is how you determine if it's called in startup or shut down case?

Reply:
Also wait I think the logic here is inverted unless I am very tired and reading it wrong. 

`has_old_buffers == FLB_TRUE` is the startup case. In which case we should print the retry message for all cases. 

I think ``has_old_buffers == FLB_FALSE` is the shutdown case where we want to print the custom message about the buffer remaining on the filesystem